PayloadPkg/Include/PreOsBoot.h

```diff
@@ -0,0 +1,33 @@
+/** @file
+
+  Copyright (c) 2019, Intel Corporation. All rights reserved.<BR>
+  This program and the accompanying materials
+  are licensed and made available under the terms and conditions of the BSD License
+  which accompanies this distribution.  The full text of the license may be found at
+  http://opensource.org/licenses/bsd-license.php.
+
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+**/
+
+#ifndef _PRE_OS_BOOT_H_
+#define _PRE_OS_BOOT_H_
+
+/** 
+  Pre-OS Boot structure corresponding to information being
+  passed to it by Slim Bootloader.
+**/
+typedef struct
+{
+  UINT32 eaxGpr;    /**< EAX General Purpose Register. */
+  UINT32 ecxGpr;    /**< ECX General Purpose Register. */
+  UINT32 edxGpr;    /**< EDX General Purpose Register. */
+  UINT32 ebxGpr;    /**< EBX General Purpose Register. */
+  UINT32 espGpr;    /**< ESP General Purpose Register. */
+  UINT32 ebpGpr;    /**< EBP General Purpose Register. */
+} PRE_OS_BOOT_PARAMS;
```

Comment:
This structure is not finalized yet; we may need to pass in additional registers (e.g. ESI, EDI, EIP, etc.).

Reply:
I would suggest to add these.  Aslo we can consider to match PUSHAD/POPAD.  So that the preOS can just use a simple POPAD/RET to jump to real OS image entry.

Reply:
And if it is just a single structure,  I think we can define it in existing header file.

Reply:
How about using CPU_BOOT_STATE that we ever used for trusty boot?  Of cause we need put it to a common place, maybe have a more generic name for this solution.